generator/types.go

```diff
@@ -368,11 +401,30 @@ func (m *Model) CtorRetVars() string {
 }
 
 // SetFields sets all the children fields and their model to the current model.
+// It also finds the primary key and sets it in the model.
 func (m *Model) SetFields(fields []*Field) {
+	var fs []*Field
+	var id *Field
 	for _, f := range fields {
+		if f.Type == BaseModel {
+			m.ID = f.Fields[0]
+			m.ID.Type = defaultIDType
+			id = f
+		}
+
 		f.Model = m
-		m.Fields = append(m.Fields, f)
+		if f.IsPrimaryKey() {
+			id = f
```

Comment:
In this very method the pk is only added once. So if there is any other PK is not added as field.